### PR TITLE
Feature/create game instance

### PR DESCRIPTION
**Pull request by ArnaudPoletto**

### CODE REVIEW

File: app/src/main/java/sdp/moneyrun/MenuActivity.java [hunk 1]
```diff
@@ -280,5 +288,29 @@ public PopupWindow onButtonShowPopupWindowClick(View view, Boolean focusable, in
 //        }
 //        //TODO: put player in the database with playerId as primary key
 //    }
-    
+
+    public void onButtonShowQuestionPopupWindowClick(View view, Boolean focusable, int layoutId, Riddle riddle) {
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/MenuActivity.java:292]: I assume this happened when you resolved conflicts, but this should not be here.

File: app/src/main/java/sdp/moneyrun/menu/JoinGameImplementation.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    private void createPlayerCountNameInfoDisplay(GameRepresentation gameRepresentation, TableRow gameRow){
+        TextView playerNumberView = new TextView(activity);
+        String playerNumberText = String.format((activity.getResources().getString(R.string.game_player_number_display)),
+                gameRepresentation.getPlayerCount(),
+                gameRepresentation.getMaxPlayerCount());
+        playerNumberView.setText(playerNumberText);
+        gameRow.addView(playerNumberView);
+    }
+
+    private void joinLobbyFromJoinButton(GameRepresentation gameRepresentation){
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/menu/JoinGameImplementation.java:223]: This button should be tested

File: app/src/main/java/sdp/moneyrun/Game.java [hunk 3]
```diff
@@ -2,36 +2,61 @@
...
 
-    public static void startGame(Game game) {
-        game.startGame();
+    public String getGameId(){
+        return gameId;
     }
 
-    // Launched when create game button is pressed
-    public void startGame() {
+    public String getName(){
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/Game.java:40]: Very easy and quick tests that would improve coverage.

File: app/src/main/java/sdp/moneyrun/Game.java [hunk 4]
```diff
@@ -2,36 +2,61 @@
...
+        return name;
+    }
 
+    public int getPlayerCount(){
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/Game.java:44]: Same as before.

File: app/src/main/java/sdp/moneyrun/Game.java [hunk 5]
```diff
@@ -2,36 +2,61 @@
...
+        return players.size();
     }
 
-    public boolean askPlayer(Player player, Riddle riddle) {
+    public int getMaxPlayerCount(){
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/Game.java:48]: Same as before.

**Review by Alvaro2112 — CHANGES_REQUESTED** [review]

File: app/src/main/java/sdp/moneyrun/MenuActivity.java [hunk 6]
```diff
@@ -235,35 +222,6 @@ public void onButtonSwitchToUserProfileActivity(View view) {
         setPutExtraArguments(playerProfileIntent);
         startActivity(playerProfileIntent);
     }
-
-    public void onButtonShowJoinGamePopupWindowClick(View view, Boolean focusable, int layoutId) {
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/MenuActivity.java:230]: Should this have been removed?

File: app/src/main/java/sdp/moneyrun/Game.java (hunk 4 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/Game.java:44]: done

File: app/src/main/java/sdp/moneyrun/Game.java (hunk 3 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/Game.java:40]: done

File: app/src/main/java/sdp/moneyrun/MenuActivity.java (hunk 1 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/MenuActivity.java]: fixed

File: app/src/main/java/sdp/moneyrun/MenuActivity.java (hunk 6 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/MenuActivity.java:230]: indeed, done

File: app/src/main/java/sdp/moneyrun/menu/JoinGameImplementation.java (hunk 2 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/menu/JoinGameImplementation.java:223]: There may be 0 or more buttons depending on the database, and some of the existing buttons may be unclickable. This is hard to test without database emulators.

File: app/src/main/java/sdp/moneyrun/Game.java (hunk 5 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/Game.java:48]: done

File: app/src/main/java/sdp/moneyrun/menu/JoinGameImplementation.java (hunk 2 above)
Comment (Alvaro2112) [reply app/src/main/java/sdp/moneyrun/menu/JoinGameImplementation.java:223]: Fair enough

**Review by gauthierboeshertz — APPROVED** [review]

**Review by Alvaro2112 — APPROVED** [review]

**Pull request merged**